options/internal/generic/essential.cpp

```diff
@@ -189,14 +189,21 @@ void *memset(void *dest, int byte, size_t count) {
 // --------------------------------------------------------------------------------------
 
 void *memmove(void *dest, const void *src, size_t size) {
-	char *dest_bytes = (char *)dest;
-	char *src_bytes = (char *)src;
-	if(dest_bytes < src_bytes) {
+	// Use uintptr_t for pointer comparisons because otherwise it's undefined behaviour
+	// when dest and src point to different objects.
+	uintptr_t udest = (uintptr_t)dest;
```

Comment:
Might be nice to use C++-style `uintptr_t udest = uintptr_t(dest)` here.

Reply:
I think if we're gonna prefer one style over the other it should be enforced somewhere (e.g in clang-tidy), we do use C-style casts in many places

Reply:
And also `uintptr_t(dest)` is still doing the same logic as c-style cast so it isn't really any better from a logic standpoint (unlike the proper c++ style casts like static_cast/reinterpret_cast/const_cast that don't just try all the casts after each one like c-style casts do).

Reply:
Uses `reinterpret_cast` now